### PR TITLE
add "query()" which generates callback events for each bistable input

### PR DESCRIPTION
**Pull request by sumnerboy12**
untested but should hopefully be close - I will test once I get my workbench setup again in the new house!

### CODE REVIEW

**Comment by sumnerboy12** [comment]
All I need to do is add a command handler to the State and Security Monitor firmware so if you send `{ "queryAll": true }` it will set a flag, then in the next processing `loop()` call `input.query()` for each MCP and reset the flag.

File: src/OXRS_Input.cpp [hunk 1]
```diff
@@ -102,6 +102,44 @@ void OXRS_Input::process(uint8_t id, uint16_t value)
...
+      // Only makes sense to publish the current state for CONTACT, SECURITY and SWITCH inputs
+      switch (type) 
+      {
+        case CONTACT:
+        case SWITCH:
+          // Ignore if we are in the middle of debounce checking
+          if (state == IS_HIGH)
+            _callback(id, i, type, HIGH_EVENT);
+          else if (state == IS_LOW)
+            _callback(id, i, type, HIGH_EVENT);
```
Comment (moinmoin-sh) [inline src/OXRS_Input.cpp]:
```C
 _callback(id, i, type, HIGH_EVENT);  
```
shouldn't be this `LOW_EVENT` ?

**Review by moinmoin-sh — CHANGES_REQUESTED** [review]

File: src/OXRS_Input.cpp (hunk 1 above)
Comment (sumnerboy12) [reply src/OXRS_Input.cpp]: Yes! Good catch, thanks for having a look.

**Comment by sumnerboy12** [comment]
Tested this on my KNX firmware and it is working nicely. Once you approve I will merge and update the State Monitor and Security Monitor firmware.

**Comment by sumnerboy12** [comment]
Here are the changes to my KNX state monitor firmware (which I will copy to the others if you are happy)... https://github.com/sumnerboy12/OXRS-BJ-StateMonitor-KNX-FW/pull/6

**Review by moinmoin-sh — APPROVED** [review]
Looks good to me now

**Pull request merged**